source/common/http/http2/codec_impl.cc

```diff
@@ -2299,14 +2299,14 @@ Http::Status ServerConnectionImpl::dispatch(Buffer::Instance& data) {
   RETURN_IF_ERROR(protocol_constraints_.checkOutboundFrameLimits());
   if (should_send_go_away_and_close_on_dispatch_ != nullptr &&
       should_send_go_away_and_close_on_dispatch_->shouldShedLoad()) {
-    ConnectionImpl::goAway();
+    ConnectionImpl::shutdownNotice();
     sent_go_away_on_dispatch_ = true;
     return envoyOverloadError(
         "Load shed point http2_server_go_away_and_close_on_dispatch triggered");
   }
   if (should_send_go_away_on_dispatch_ != nullptr && !sent_go_away_on_dispatch_ &&
       should_send_go_away_on_dispatch_->shouldShedLoad()) {
-    ConnectionImpl::goAway();
+    ConnectionImpl::shutdownNotice();
```

Comment:
If this is necessary could you add a comment to explain why then to avoid the future developer/reader be confused? thanks. :)